fix endpoint url paths

leahoswald opened the pull request:
The endpoint url paths combined with the rest of the code have a slash to much leading to an URL like https://api.home-connect.com///security/oauth/authorize this breaks because the home connect api doesn't respond properly on this URL.

leahoswald committed:
fix endpoint url paths

The endpoint url paths combined with the rest of the code have a slash to much leading to an URL like https://api.home-connect.com///security/oauth/authorize this breaks because the home connect api doesn't respond properly on this URL.

DavidMStraub commented:
Hi, thanks - I wonder why this didn't cause me issues before.

I noticed though that the code contains both `f"{self.host}{ENDPOINT_TOKEN}"` and `f"{self.host}/{ENDPOINT_AUTHORIZE}"`, so removing the slash will not work either. I suggest to change it to `f"{self.host}{ENDPOINT_SOMETHING}"` everywhere and restore the slash in the constants.

leahoswald committed:
revert URL path change and fix inconsitent usage in url building instead

removing lstrip of "/"  from one URL building and additional "/" from
two other parts fixing broken URLs

leahoswald commented:
Added another commit that should fix the problems in the code. With this change it works for me too.

DavidMStraub commented:
Thanks!

Sorry for being pedantic (more pedantic than when writing the code I guess :wink:) but can you please undo the change to `url = f"{self.host}/{path.lstrip('/')}"`? The point is that the existing implementation was robust to having or not having a leading slash in `path`, while after your change it would break if `path` does not start with a slash.

leahoswald committed:
revert removing lstrip of "/"

leahoswald commented:
No problem, fixed now. But I want to add that I'm not quite sure if it is good to have one part of the code where the / is removed and then added again so it works with and without slash in path and other parts where we rely on the existence of the slash in path.

DavidMStraub commented:
Thanks!
